src/__tests__/components/frontend-engine/yup/custom-conditions.spec.ts

```diff
@@ -17,4 +19,74 @@ describe("validateUinfin", () => {
 			expect(TestHelper.getError(() => schema.validateSync("S1234567A")).message).toBe("error");
 		});
 	});
+
+	describe("equalsField", () => {
+		it.each`
+			type        | field1    | field2
+			${"string"} | ${"text"} | ${"text"}
+			${"number"} | ${10}     | ${10}
+		`("should not throw error if both inputs are same $type", ({ field1, field2 }) => {
+			const schema = YupHelper.buildSchema({
+				field1: {
+					schema: Yup.string(),
+					validationRules: [{ required: true }],
+				},
+				field2: { schema: Yup.string(), validationRules: [{ equalsField: "field1" }] },
+			});
+			expect(() => schema.validateSync({ field1, field2 })).not.toThrowError();
+		});
+
+		it.each`
+			type        | field1    | field2
+			${"string"} | ${"text"} | ${"test"}
+			${"number"} | ${10}     | ${11}
+		`("should throw error if both inputs are not same $type", ({ field1, field2 }) => {
+			const ERROR_MESSAGE = "test error message";
+			const schema = YupHelper.buildSchema({
+				field1: {
+					schema: Yup.string(),
+					validationRules: [{ required: true }],
+				},
+				field2: {
+					schema: Yup.string(),
+					validationRules: [{ equalsField: "field1", errorMessage: ERROR_MESSAGE }],
+				},
+			});
+			expect(TestHelper.getError(() => schema.validateSync({ field1, field2 })).message).toBe(ERROR_MESSAGE);
+		});
+
+		it.each`
+			type             | field1       | field2
+			${"array"}       | ${["Apple"]} | ${["Apple"]}
+			${"array empty"} | ${[]}        | ${[]}
+		`("should not throw error if both inputs are same $type", ({ field1, field2 }) => {
+			const schema = YupHelper.buildSchema({
+				field1: {
+					schema: Yup.array(),
+					validationRules: [{ required: true }],
+				},
+				field2: { schema: Yup.array(), validationRules: [{ equalsField: "field1" }] },
+			});
+			expect(() => schema.validateSync({ field1, field2 })).not.toThrowError();
+		});
+
+		it.each`
+			type                | field1                 | field2
+			${"two one array"}  | ${["Apple", "Berry"]}  | ${["Apple", "Cherry"]}
+			${"two zero array"} | ${["Apple", "Cherry"]} | ${[]}
+		`("should throw error if both inputs are not same $type", ({ field1, field2 }) => {
+			const ERROR_MESSAGE = "test error message";
+			const schema = YupHelper.buildSchema({
+				field1: {
+					schema: Yup.array(),
+					validationRules: [{ required: true }],
+				},
+				field2: {
+					schema: Yup.array(),
+					validationRules: [{ equalsField: "field1", errorMessage: ERROR_MESSAGE }],
+				},
+			});
+			expect(TestHelper.getError(() => schema.validateSync({ field1, field2 })).message).toBe(ERROR_MESSAGE);
+		});
+	});
```

Comment:
- can put your test cases in `src/__tests__/components/frontend-engine/yup/yup-helper.spec.ts` under `mapRules`, there you can add a more extensive list of tests against strings, numbers, arrays and objects
- the tests in this file are extra tests to assert any internal logic that are hard to test in the yup-helper test suite, you don't need to add test cases here